### DISCUSSION TITLE
Handling python_packages objects:

### DISCUSSION THREAD

**Discussion by tatsuya-takakuwa** [DanWBR/dwsim#524] [General]
Please tell me how to load input parameters of python_packages (e.g. PEM models) in an existing simulation file and how to change the values and re-store them.

**Comment by DanWBR** [comment]
@tatsuya-takakuwa I don't understand, could you please give me more details?

**Reply by tatsuya-takakuwa** [reply]
Sorry for the lack of clarity.
I am coupling an external Python program with DWSIM to modify input variables on DWSIM to perform data generation for deep learning.
Using Automation3, I have successfully loaded a flowsheet from an existing dwxmz file.
We are also able to output unit and stream values by object name.
PEM model as well, each parameter can be retrieved with .GetPropertyValue.
However, using .SetPropertyValue("inputpameter_name",value) to change the value set on Python does not change the value in the original dwxmz file.
Please let me know if there is a correct way to do this.

**Reply by DanWBR** [reply]
see here how the calculation is done -> you can set the properties directly by accessing the InputParameters and OutputParameters dictionaries: https://github.com/DanWBR/dwsim/blob/windows/DWSIM.UnitOperations/UnitOperations/CleanEnergies/PEMFC_Amphlett.vb#L113

**Reply by tatsuya-takakuwa** [reply]
Thank you. However, I still do not understand the compatibility between vbnet and python.
How do I put together the Python code to access the input and output dictionaries?

**Reply by DanWBR** [reply]
it is as simple as this:

`current = mypemfc.InputParameters['A']`

**Reply by tatsuya-takakuwa** [reply]
Thank you very much! It worked fine.
I ran a SOLVE calculation on a CalucurationFlowsheet and saved the results of the flowsheet object, but the output does not reflect the results of the calculation.

How do I run Solve?

**Reply by DanWBR** [reply]
see this https://gist.github.com/DanWBR/fe979513ba32627b47b02888061b69f4

**Reply by tatsuya-takakuwa** [reply]
Thank you very much!
It worked without a hitch! Thanks to you, we can design a green energy production facility.
